core/controllers/access_validators.py

```diff
@@ -80,6 +80,40 @@ def get(self) -> None:
             raise self.NotFoundException
 
 
+class ClassroomsPageAccessValidationHandler(
+    base.BaseHandler[Dict[str, str], Dict[str, str]]
+):
+    """Validates access to classrooms page."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS: Dict[str, str] = {}
+    HANDLER_ARGS_SCHEMAS: Dict[str, Dict[str, str]] = {
+        'GET': {}
+    }
+
+    @acl_decorators.open_access
+    def get(self) -> None:
+        """Retrieves information about classrooms.
+
+        Raises:
+            PageNotFoundException. The multiple classrooms
+                feature is not enabled.
+            PageNotFoundException. No public classrooms are present.
+        """
+        classrooms = classroom_config_services.get_all_classrooms()
+        have_atleast_one_public_classroom = False
+        for classroom in classrooms:
+            if classroom.is_published:
+                have_atleast_one_public_classroom = True
+                break
+
+        if not feature_flag_services.is_feature_flag_enabled(
+            feature_flag_list.FeatureNames.ENABLE_MULTIPLE_CLASSROOMS.value,
+            user_id=self.user_id
+        ) or (not have_atleast_one_public_classroom and not constants.DEV_MODE):
```

Comment:
Too many not-s, I find it difficult to understand what's going on here.

Also, feature flag check might be moved to the top of the function. Then it's `if not has_public_classrooms and not constants.DEV_MODE`, which is fine

Reply:
Done.